torchani/utils.py

```diff
@@ -192,7 +192,7 @@ def sae(self, species):
             intercept = self.self_energies[-1]
 
         self_energies = self.self_energies[species]
-        self_energies[species == torch.tensor(-1, device=species.device)] = torch.tensor(0, device=species.device, dtype=self_energies.dtype)
+        self_energies[species == torch.tensor(-1, device=species.device)] = torch.tensor(0, device=species.device, dtype=torch.double)
```

Comment:
@zasdfgbnm `self_energies.dtype` should be `double`, but `float32` is enforced in `comp6.py`, is there any particular reason for that?

Reply:
@farhadrgh No, there is no reason. I guess it is just randomly picking a dtype.

Reply:
@farhadrgh Also, there are some trick here, energy shifter is a module, so if you put it inside
`model = torch.nn.Sequential(aev_computer, nn, energy_shifter)` and then `model.to(torch.float)` it will also be cast into float. I don't have a good idea on how to solve that.

Reply:
I fixed the `dtype` in `comp6.py` (6930879), this will let the tests pass peacefully